assets/vsphere_problem_detector/12_prometheusrules.yaml

```
@@ -0,0 +1,25 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: vsphere-problem-detector
+  namespace: openshift-cluster-storage-operator
+  labels:
+    role: alert-rules
+spec:
+  groups:
+    - name: vsphere-problem-detector.rules
+      rules:
+      - alert: VSphereOpenshiftNodeHealthFail
+        expr:  vsphere_node_check_errors == 1
```

What type is this metric? A counter or gauge? Quick search could not find it in the operator.

---

It's gauge now, https://github.com/openshift/vsphere-problem-detector/pull/24
(used to be counter yesterday, hard to alert on it).

---

One small comment to improve the alerting rule: in the current version, a failed scrape by Prometheus would resolve the alert if it was firing previously.
To protect against it, you can use `min_over_time()` like this

```suggestion
        expr:  min_over_time(vsphere_node_check_errors[5m]) == 1
```

see https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0

---

> (used to be counter yesterday, hard to alert on it).

Nice thanks! 

Agreed with Simons suggestion, otherwise looks good to me.

---

why `min_over_time` - shouldn't this be `max_over_time`?  I would think if a scrape failed and a value is missing on time t1 then it would replace with some kind of sentinel value (`0`?  - I don't know prometheus very well and how it fills the holes in the data) .

---

If the target is down (up == 0) then Prometheus will mark the `vsphere_node_check_errors` metric as stale (meaning it doesn't exist anymore). On the next evaluation of the alerting rule, the result from the rule's expression would be "no data" so Prometheus will consider that the alert is resolved.